internal/users/domain/user_factory.go

```diff
@@ -0,0 +1,18 @@
+package domain
+
+import "time"
+
+type UserFactory struct{}
+
+func NewUserFactory() *UserFactory {
+	return &UserFactory{}
+}
+
+func (f UserFactory) CreateNewUser(user *NewUser) *User {
```

Comment:
I think arguments must be username, email, password .

Reply:
appreciate it.